src/parse/ast.rs

```diff
@@ -1,11 +1,143 @@
-//! Defines types of the abstract syntax tree
-//!
-
 /// Top level type. Is returned by `parse`.
 #[derive(Debug, Clone)]
 pub enum Query {
-    Select,
-    Insert,
-    Update,
-    Delete
+    DefStmt(DefStmt),
+    ManipulationStmt(ManipulationStmt)
+}
+
+#[derive(Debug, Clone)]
+pub enum DefStmt {
+    Create(CreateStmt),
+    Alter(AltStmt),
+    Drop(DropStmt)
+}
+
+#[derive(Debug, Clone)]
+pub enum ManipulationStmt {
+	Update(UpdateStmt),
+	Select(SelectStmt),
+	Insert(InsertStmt),
+	Delete(DeleteStmt)
+}
+
+#[derive(Debug, Clone)]
+pub enum CreateStmt {
+    Table(CreateTableStmt),
+    // View
+}
+
+#[derive(Debug, Clone)]
+pub enum AltStmt {
+    Table(AlterTableStmt)
+    //Column(String)
+    //View(String)
+}
+
+#[derive(Debug, Clone)]
+pub enum DropStmt {
+    Table(String)
+    //Column(String)
+    //Index(String)
+    //Database(String)
+}
+
+#[derive(Debug, Clone)]
+pub struct CreateTableStmt {
+    pub tid: String,
+    pub cols: Option<Vec<CreateColumn>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct CreateColumn {
+    pub id: String,
+    pub datatype: DType,
+}
+
+#[derive(Debug, Clone)]
+pub struct AlterTableStmt {
+	pub id: String,
+	pub op: AlterOp
+}
+
+#[derive(Debug, Clone)]
+pub enum AlterOp {
+	Add(CreateColumn),
+	Drop(String),
+	Alter(CreateColumn)
+}
+
+#[derive(Debug, Clone)]
+pub struct UpdateStmt {
+	pub tid: String,
```

Comment:
`tid` probably stands for "table id". In some types above I've seen the name `id` for the table id field. Would be nice to call it the same everywhere.